lib/commons/standards/implicit-html-roles.js

```diff
@@ -131,6 +120,22 @@ const implicitHtmlRoles = {
         return 'slider';
       case 'search':
         return !suggestionsSourceElement ? 'searchbox' : 'combobox';
+
+      case 'button':
+      case 'image':
+      case 'reset':
+      case 'submit':
+        return 'button';
+
+      case 'text':
+      case 'tel':
+      case 'url':
+      case 'email':
+      case '':
```

Comment:
According to https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#htmlattrdeflist the `list` attribute is valid for more than what we have, so `input type="date" list="someId"` should be considered a combobox instead of a textbox. We should also add these to the case list to make it more complete

Reply:
It's not a combobox according to https://www.w3.org/TR/html-aria/#el-input-text-list